### ISSUE TITLE
[SUGGESTION] Prefabricated turrets

### ISSUE THREAD

**Issue by zorozeenee** [tsunamayo/Starship-EVO#2526]
Building the same type of turret on your ship over and over again can get very tedious.
It would be nice to be able place turrets that you already have saved in your blueprints on your ship.

There are several ways of doing this

1. Pressing f on a rotator would open a menu where you could spawn your turret and tell it which direction it should face.
2. You could add a new tool where you can point at a flat block on a ship (turret) you already made and spawned then right click the rotator to attach it

**Comment by tsunamayo** [comment]
Yes it is a valid suggestion. The downside of this is that turret are expensive game-wise, so I dont want player to mass spawn them too easily. So I will have to get a turret token kind of system...

**Comment by Kaiser-Indrasil** [comment]
I have dozens of turret designs in mind I'm gonna build once there is some kind of blueprint copy-pasting in place. So hold on tight!

**Comment by Majikmonster** [comment]
This is a great feature to have. Starmade did exactly this. It saved so much time on larger creations, even if they only had a couple of turrets. Some turrets can be very complex design-wise, so building them more than once becomes a chore very quickly.

The preview hologram that the docking rings create when you use the connector tool would be best to add with a feature like this.

**Comment by zorozeenee** [comment]
> Yes it is a valid suggestion. The downside of this is that turret are expensive game-wise, so I dont want player to mass spawn them too easily. So I will have to get a turret token kind of system...

Rebuilding 30 turrets is expensive for my brain

**Comment by esauftgz** [comment]
I think sacrificing QoL for possible abuse is a bad idea.
After all people who really want 50 turrets will do it in time, but it only inconveniences builders with less time to ensure everyone single one of their turrets are identical for aesthetics' sake, even if its just a small number like 2-6 turrets.

**Comment by Calvert4096** [comment]
Definitely in favor of quality-of-life features.  I think a generic "sub-assembly" blueprint tool would cover this.  If it's desired to prevent laggy builds, then give server admins tools to put limits on entities/ mechanisms/computer elements etc. per player.
